rpc/beefy/get_finalized_head.go

```diff
@@ -0,0 +1,33 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2021 Snowfork
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package beefy
+
+import (
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+// GetFinalizedHead returns the hash of the latest BEEFY block
+func (b *Beefy) GetFinalizedHead() (types.Hash, error) {
+	var res string
+
+	err := b.client.Call(&res, "beefy_getFinalizedHead")
+	if err != nil {
```

Comment:
Some of these lines can be easily covered by unit tests if we would use a client mock. What do you think @vgeddes ?

Reply:
If you mean mocking out the low-level call to `beefy_getFinalizedHead`, I don't think there is a testing harness in GSRPC to allow mocking at that level. We'd have to implement one first. Additionally, since `beefy_getFinalizedHead` isn't enabled by default in Substrate builds, we can't really test it using the existing docker E2E tests.

In general for mock-driven RPC unit tests, I'm trying to understand what value they provide, since we're mocking out the stuff that actually matters - the interactions with the server.

Reply:
No, I was referring mainly to the implementation here, basically covering the lines that are mentioned by codecov. In any case, we can probably skip this now since the coverage diff is minimal and we also have to improve the testing of the other RPCs.